src/components/schedule-visualizer/tests/robot-image-icon.test.tsx

```diff
@@ -0,0 +1,45 @@
+import { mount } from 'enzyme';
+import React from 'react';
+import { makeRobot } from '../../../mock/utils';
+import ColorManager from '../colors';
+import RobotImageIcon from '../robot-image-icon';
+
+describe('RobotImageIcon', () => {
+  const colorManager = new ColorManager();
+  // TextEncoder is not available in node
+  colorManager.robotColor = jest.fn(async () => 'black');
+  colorManager.robotColorFromCache = jest.fn(() => 'black');
+
+  const iconPath = 'testIcon';
+
+  test('renders correctly', async () => {
+    const root = mount(
+      <svg>
+        <RobotImageIcon
+          robot={makeRobot()}
+          footprint={1}
+          colorManager={colorManager}
+          dispatchIconError={jest.fn()}
+          iconPath={iconPath}
+        />
+      </svg>,
+    );
+    expect(root).toMatchSnapshot();
```

Comment:
-> `expect(toJson(root)).toMatchSnapshot();` you need to convert the container to Json with `enzyme-to-json` library first, an then compare with the snapshot. If you don't make that conversion the snapshot will be an empty object. You can check that here `src/components/schedule-visualizer/tests/__snapshots__/robot-default-icon.test.tsx.snap`

exports[`RobotDefaultIcon renders correctly 1`] = `ReactWrapper {}`;

Reply:
You are right, totally missed that. There is also `root.html()` to get the rendered output, should I use that or convert it to json format?

Reply:
Yeah, this is what each command generates. My personal opinion is that it's easier to read what the component is rendering by converting it to JSON.

### root.html():

```
exports[`RobotImageIcon renders correctly 1`] = `"<svg><g><defs><radialGradient id=\\"RobotImageIcon-0-shadow\\"><stop offset=\\"0%\\" stop-color=\\"#00000080\\"></stop><stop offset=\\"70%\\" stop-color=\\"#00000040\\"></stop><stop offset=\\"90%\\" stop-color=\\"#00000010\\"></stop><stop offset=\\"100%\\" stop-color=\\"#00000000\\"></stop></radialGradient><radialGradient id=\\"RobotImageIcon-0-shadow-conflict\\"><stop offset=\\"0%\\" stop-color=\\"#f4433680\\"></stop><stop offset=\\"70%\\" stop-color=\\"#f4433640\\"></stop><stop offset=\\"90%\\" stop-color=\\"#f4433610\\"></stop><stop offset=\\"100%\\" stop-color=\\"#f4433600\\"></stop></radialGradient></defs><circle id=\\"shadow\\" r=\\"1.3\\" fill=\\"url(#RobotImageIcon-0-shadow)\\"></circle><g transform=\\"translate(-1 -1)\\"><image href=\\"testIcon\\" height=\\"2\\" width=\\"2\\"></image></g></g></svg>"`;
```


### toJson(root):

```
exports[`RobotImageIcon renders correctly 1`] = `
<svg>
  <ForwardRef
    colorManager={
      ColorManager {
        "_robotColorCache": Object {},
        "conflictHighlight": "#f44336",
        "robotColor": [MockFunction],
        "robotColorFromCache": [MockFunction],
      }
    }
    dispatchIconError={[MockFunction]}
    footprint={1}
    iconPath="testIcon"
    robot={
      Object {
        "battery_percent": 100,
        "location": Object {
          "level_name": "",
          "t": Object {
            "nanosec": 0,
            "sec": 0,
          },
          "x": 0,
          "y": 0,
          "yaw": 0,
        },
        "mode": Object {
          "mode": 0,
        },
        "model": "BaseModel",
        "name": "",
        "path": Array [],
        "task_id": "",
      }
    }
  >
    <g>
      <defs>
        <Component
          id="RobotImageIcon-0-shadow"
        >
          <radialGradient
            id="RobotImageIcon-0-shadow"
          >
            <stop
              offset="0%"
              stopColor="#00000080"
            />
            <stop
              offset="70%"
              stopColor="#00000040"
            />
            <stop
              offset="90%"
              stopColor="#00000010"
            />
            <stop
              offset="100%"
              stopColor="#00000000"
            />
          </radialGradient>
        </Component>
        <Component
          id="RobotImageIcon-0-shadow-conflict"
        >
          <radialGradient
            id="RobotImageIcon-0-shadow-conflict"
          >
            <stop
              offset="0%"
              stopColor="#f4433680"
            />
            <stop
              offset="70%"
              stopColor="#f4433640"
            />
            <stop
              offset="90%"
              stopColor="#f4433610"
            />
            <stop
              offset="100%"
              stopColor="#f4433600"
            />
          </radialGradient>
        </Component>
      </defs>
      <circle
        fill="url(#RobotImageIcon-0-shadow)"
        id="shadow"
        r={1.3}
      />
      <g
        transform="translate(-1 -1)"
      >
        <image
          height={2}
          href="testIcon"
          onError={[Function]}
          width={2}
        />
      </g>
    </g>
  </ForwardRef>
</svg>
`;


```